resolwe/flow/managers/listener.py

```diff
@@ -501,7 +501,14 @@ def handle_download_finished(self, obj):
                 obj, {ExecutorProtocol.RESULT: ExecutorProtocol.RESULT_ERROR},
             )
         else:
-            query.update(status=StorageLocation.STATUS_DONE)
+            # Add files to the downloaded storage location.
+            storage_location = query.get()
+            default_storage_location = (
+                storage_location.file_storage.default_storage_location
+            )
+            storage_location.files.add(*default_storage_location.files.all())
+            storage_location.status = StorageLocation.STATUS_DONE
```

Comment:
I think that `<m2m>.add` executes a query to add related objects, so you can keep `update` here instead of `save`.